src/Nethermind/Nethermind.Sockets/SocketClient.cs

```diff
@@ -15,15 +15,23 @@ public class SocketClient<TStream> : ISocketsClient where TStream : Stream, IMes
 
     protected readonly TStream _stream;
     protected readonly IJsonSerializer _jsonSerializer;
+    private readonly int _maxRequestSize;
 
     public string Id { get; } = Guid.NewGuid().ToString("N");
     public string ClientName { get; }
 
-    public SocketClient(string clientName, TStream stream, IJsonSerializer jsonSerializer)
+    public SocketClient(string clientName, TStream stream, IJsonSerializer jsonSerializer) :
+        // Optional maxRequestSize break some tests.
+        this(clientName, stream, jsonSerializer, MAX_REQUEST_BODY_SIZE_FOR_ENGINE_API)
+    {
+    }
+
+    public SocketClient(string clientName, TStream stream, IJsonSerializer jsonSerializer, int maxRequestSize)
```

Comment:
No need for 2 constructors?
```suggestion
    public SocketClient(string clientName, TStream stream, IJsonSerializer jsonSerializer, int maxRequestSize = MAX_REQUEST_BODY_SIZE_FOR_ENGINE_API)
```